openpype/plugins/publish/validate_unique_subsets.py

```diff
@@ -49,17 +53,41 @@ def process(self, context):
                                  "{}".format(instance.name))
                 continue
 
-            instance_per_asset_subset[(asset, subset)].append(instance)
+            # for cases where multiple versions under same subset are published
+            version = instance.data.get("version")
+            if version is None:
+                # this might happen when workfile version sync is disabled
+                # we still need to validate cases where multiple "next version"
+                # instances are published under same subset
+                self.log.info("Instance found without `version` data: "
+                              "{}".format(instance.name))
+            else:
+                # check if asset and subset combination is unique
+                # just to inform the user with warning about the case
+                if any([
+                    asset == a and subset == s
+                    for a, s, _ in instance_per_asset_subset.keys()
+                ]):
+                    self.log.warning(
+                        "Instance found with non unique `asset` and "
+                        "`subset` data: {}".format(instance.name)
+                    )
```

Comment:
I wouldn't do this check in this for loop - it's basically much slower like this because you iterate over all per entry.